### PR TITLE
IDEX-2965: removed the ability to import project without type and spe…

### PR DESCRIPTION
**Pull request by akorneta**
…cified configs

### CODE REVIEW

File: platform-api/che-core-api-project/src/main/java/org/eclipse/che/api/project/server/ProjectService.java [hunk 1]
```diff
@@ -811,55 +812,52 @@ private VirtualFileEntry getVirtualFile(String workspace, String path, boolean f
...
-            if (newProject != null) {
-                visibility = newProject.getVisibility();
-                RunnersDescriptor descriptor = newProject.getRunners();
-                if (descriptor != null) {
-                    replaceDefaultRunnerPath(descriptor);
-                }
-                projectConfig = DtoConverter.fromDto2(newProject, projectTypeRegistry);
-            }
+            visibility = newProject.getVisibility();
+            RunnersDescriptor descriptor = newProject.getRunners();
```
Comment (mshaposhnik) [inline platform-api/che-core-api-project/src/main/java/org/eclipse/che/api/project/server/ProjectService.java]: Never can be null?

File: platform-api/che-core-api-project/src/main/java/org/eclipse/che/api/project/server/ProjectService.java [hunk 2]
```diff
@@ -874,6 +872,13 @@ private ImportResponse configureProject(ImportProject importProject, FolderEntry
                 postImportProjectHandler.onProjectImported(project.getBaseFolder());
             }
         } catch (ConflictException | ForbiddenException | ServerException | NotFoundException e) {
+            if (newProject.getDescription() != null
+                || newProject.getAttributes() != null
+                || newProject.getRunners() != null
+                || newProject.getBuilders() != null
+                || newProject.getMixinTypes() != null) {
+                throw new BadRequestException("Impossible to save configurations for the project without type");
```
Comment (mshaposhnik) [inline platform-api/che-core-api-project/src/main/java/org/eclipse/che/api/project/server/ProjectService.java]: this code looks strange. how the server exception related to the given message ?

**Comment by garagatyi** [comment]
LGTM
BTW code in ProjectService is completely unclear. Code reviews should not accept such code, but I suppose it was done even before we started make code reviews by multiple people in that area of project.

**Comment by skabashnyuk** [comment]
LGTM

File: platform-api/che-core-api-project/src/main/java/org/eclipse/che/api/project/server/ProjectService.java (hunk 1 above)
Comment (akorneta) [inline platform-api/che-core-api-project/src/main/java/org/eclipse/che/api/project/server/ProjectService.java]: if it be null then this method throw npe here https://github.com/codenvy/che-core/blob/master/platform-api/che-core-api-project/src/main/java/org/eclipse/che/api/project/server/ProjectService.java#L888

File: platform-api/che-core-api-project/src/main/java/org/eclipse/che/api/project/server/ProjectService.java (hunk 2 above)
Comment (akorneta) [inline platform-api/che-core-api-project/src/main/java/org/eclipse/che/api/project/server/ProjectService.java]: yes its looks strange. But it's occurs when we try convert dto into project's configs.

**Pull request merged**